### PR TITLE
NIFI-4009: Added support for several key functions in RecordPath

### PR DESCRIPTION
**Pull request by markap14**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/filter/NotEqualsFilter.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public NotEqualsFilter(final RecordPathSegment lhs, final RecordPathSegment rhs)
     protected boolean test(final FieldValue fieldValue, final Object rhsValue) {
         final Object lhsValue = fieldValue.getValue();
         if (lhsValue == null) {
-            return rhsValue != null;
+            return false;
```
Comment (mattyb149) [inline nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/filter/NotEqualsFilter.java]: What's the reason for this change? Not saying it's wrong, just trying to understand it :)

Comment (markap14) [inline nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/filter/NotEqualsFilter.java]: I've seen both ways of handling this case in different platforms, where you're comparing null to null. null == null is fairly straight-forward, but null != null gets a bit weird... some platforms return true, some return false. Originally I had it returning true in such cases. But that means that we had to do some weird stuff in PredicatePath, which caused a lot of other problems when we added functions.... so I changed that to make PredicatePath work a little more cleanly. There's a pretty big blurb of inline comments in there about how we were handling it before vs now.

**Comment by mattyb149** [comment]
+1 LGTM, ran full build with unit tests and contrib-check, and tested a few functions on a NiFi instance. Thanks much! merging to master
